### PR TITLE
Issue/10638 material sharing settings

### LINKED ISSUE #10638: Implement Material theme on Sharing Settings screen
We should switch widgets used at  Sharing Settings screen to their material counterparts, remove color references where possible, and switch to using material `textAppearance` attributes.

### PR DESCRIPTION
**Pull request by khaykov**
Fixes #10638 

This PR moves Publicize screen to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

comparison_sharing

To test:
- Navigate to Publicize screen and confirm it's visual and functional integrity in both Light and Dark themes. The theme can be changed from App Settings.

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
I'm moving this PR to 14.1 since we're code freezing today.

**Comment by khaykov** [comment]
@mattmiklic Sure! I moved everything out of cards. Had to adjust margins a bit :)

[Image from Gyazo](https://gyazo.com/44613dd4a3841ff72e570f9a2af83673)
[Image from Gyazo](https://gyazo.com/8c9dc7ac2616d67e13e858c8464e928c)
[Image from Gyazo](https://gyazo.com/3ed3b7dbba1d664391f882dcedd975bf)
[Image from Gyazo](https://gyazo.com/421e920fb4c6dc1c22f6ab30fcda7b28)

**Comment by mattmiklic** [comment]
That's a nice improvement; thanks for doing that!

File: WordPress/src/main/java/org/wordpress/android/widgets/WPPrefView.java [hunk 1]
```diff
@@ -431,18 +419,14 @@ private Dialog getCheckListDialog() {
...
-                .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        SparseBooleanArray checkedItems =
-                                ((AlertDialog) dialog).getListView().getCheckedItemPositions();
-                        mListItems.setCheckedItems(checkedItems);
-                        doPrefChanged();
-                    }
+                .setPositiveButton(android.R.string.ok, (dialog, which) -> {
+                    SparseBooleanArray checkedItems1 =
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/widgets/WPPrefView.java]: Nitpick: Should we name `checkedItems1` to something else? Or maybe we could do something like this:

```
mListItems.setCheckedItems(((AlertDialog) dialog).getListView().getCheckedItemPositions());
```

_P.S: Android Studio can't find `MaterialAlertDialogBuilder` for me, so I couldn't play with it._

Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/widgets/WPPrefView.java]: Good catch! AS does it sometimes with optimizing code.

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeAccountChooserListAdapter.java [hunk 2]
```diff
@@ -38,6 +39,7 @@ public PublicizeAccountChooserListAdapter(Context context, List<PublicizeConnect
         mSelectedPosition = 0;
     }
 
+    @NotNull
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeAccountChooserListAdapter.java]: Got it! Was not sure about this :)

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/ConnectButton.java [hunk 3]
```diff
@@ -15,6 +16,7 @@
  */
 public class ConnectButton extends FrameLayout {
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/publicize/ConnectButton.java]: Agree, that's a good idea 👍

File: WordPress/src/main/java/org/wordpress/android/ui/publicize/ConnectButton.java [hunk 4]
```diff
@@ -68,4 +70,10 @@ public void setAction(ConnectAction newAction) {
             updateView();
         }
     }
+
+    @Override
+    public void setOnClickListener(@Nullable OnClickListener l) {
+        super.setOnClickListener(l);
+        mConnectButton.setOnClickListener(l);
```
[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/publicize/ConnectButton.java]: Right, we probably don't need super here. I changed the base class and removed this method all along.

**Comment by khaykov** [comment]
@oguzkocer thank you so much for checking Dark Mode PR's! Really appreciate that 🙇 I hope I addressed your concerns.

Regarding layout changes, I agree! I don't think there is a good way to approach this apart from looking at the layout.

**Pull request merged**